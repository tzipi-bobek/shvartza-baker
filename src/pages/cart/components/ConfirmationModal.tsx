interface Props {
  open: boolean;
  message: string;
  onConfirm: () => void;
  onCancel: () => void;
}

const ConfirmationModal = ({ open, message, onConfirm, onCancel }: Props) => {
  if (!open) return null;

  return (
    <div className="fixed inset-0 backdrop-blur-sm bg-white/30 z-50 flex items-center justify-center">
      <div className="bg-white p-6 rounded-lg shadow-xl max-w-sm w-full text-center space-y-4 font-serif">
        <p className="text-lg text-gray-700">{message}</p>
        <div className="flex justify-center gap-4 pt-2">
          <button
            onClick={onConfirm}
            className="btn-cremita btn-cremita-primario px-4 py-2"
          >
            Aceptar
          </button>
          <button
            onClick={onCancel}
            className="btn-cremita btn-cremita-secundario px-4 py-2"
          >
            Cancelar
          </button>
        </div>
      </div>
    </div>
  );
};

export default ConfirmationModal;
