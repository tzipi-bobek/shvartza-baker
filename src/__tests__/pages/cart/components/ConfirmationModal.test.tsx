import { render, screen, fireEvent } from '@testing-library/react';
import ConfirmationModal from '@pages/cart/components/ConfirmationModal';
import { describe, expect, test, vi } from 'vitest';

describe('ConfirmationModal', () => {
  test('does not render when "open" is false', () => {
    const { container } = render(
      <ConfirmationModal
        open={false}
        message="Are you sure?"
        onConfirm={vi.fn()}
        onCancel={vi.fn()}
      />
    );

    expect(container.firstChild).toBeNull();
  });

  test('renders the message correctly when open', () => {
    render(
      <ConfirmationModal
        open={true}
        message="Delete this item?"
        onConfirm={vi.fn()}
        onCancel={vi.fn()}
      />
    );

    expect(screen.getByText(/Delete this item\?/i)).toBeInTheDocument();
    expect(screen.getByText(/Aceptar/i)).toBeInTheDocument();
    expect(screen.getByText(/Cancelar/i)).toBeInTheDocument();
  });

  test('calls onConfirm when "Aceptar" is clicked', () => {
    const confirmMock = vi.fn();

    render(
      <ConfirmationModal
        open={true}
        message="Confirm?"
        onConfirm={confirmMock}
        onCancel={vi.fn()}
      />
    );

    fireEvent.click(screen.getByText(/Aceptar/i));
    expect(confirmMock).toHaveBeenCalled();
  });

  test('calls onCancel when "Cancelar" is clicked', () => {
    const cancelMock = vi.fn();

    render(
      <ConfirmationModal
        open={true}
        message="Confirm?"
        onConfirm={vi.fn()}
        onCancel={cancelMock}
      />
    );

    fireEvent.click(screen.getByText(/Cancelar/i));
    expect(cancelMock).toHaveBeenCalled();
  });

  test('matches snapshot when open', () => {
    const { container } = render(
      <ConfirmationModal
        open={true}
        message="Are you sure?"
        onConfirm={() => {}}
        onCancel={() => {}}
      />
    );

    expect(container).toMatchSnapshot();
  });
});
