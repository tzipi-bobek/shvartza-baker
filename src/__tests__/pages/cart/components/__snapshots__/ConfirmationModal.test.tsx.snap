// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ConfirmationModal > matches snapshot when open 1`] = `
<div>
  <div
    class="fixed inset-0 backdrop-blur-sm bg-white/30 z-50 flex items-center justify-center"
  >
    <div
      class="bg-white p-6 rounded-lg shadow-xl max-w-sm w-full text-center space-y-4 font-serif"
    >
      <p
        class="text-lg text-gray-700"
      >
        Are you sure?
      </p>
      <div
        class="flex justify-center gap-4 pt-2"
      >
        <button
          class="btn-cremita btn-cremita-primario px-4 py-2"
        >
          Aceptar
        </button>
        <button
          class="btn-cremita btn-cremita-secundario px-4 py-2"
        >
          Cancelar
        </button>
      </div>
    </div>
  </div>
</div>
`;
